sensor/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,140 @@
+    <img src="../images/tag_based_pnp_calibrator/menu4.jpg" alt="menu4" width="500">
+</p>
+
+The user can modify the `visualization options` on the right side of the `image view`. To compare the results, please set the `Marker size (m)` to `0.04` and set the `PC subsample factor` to `1`.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization_bar.jpg"  alt="visualization_bar" width="200">
+</p>
+
+After setting the options above, change the `/initial_tf` (in the `visualization options`) to `/current_tf`. By doing this, it is easier to measure the difference after the calibration.
+
+The images below show that with the calibrated transformation, the projected pointcloud aligns better with the image.
+
+<table>
+  <tr>
+    <td><img src="../images/tag_based_pnp_calibrator/init_tf.jpg" alt="init_tf" width = 500px ></td>
+    <td><img src="../images/tag_based_pnp_calibrator/calibrated_tf.jpg" alt="calibrated_tf" width = 500px ></td>
+   </tr>
+   <tr>
+    <td><p style="text-align: center;">Before Calibration.</p></td>
+    <td><p style="text-align: center;">After Calibration.</p></td>
+  </tr>
+</table>
+
+## FAQ
+
+- Why the calibrator doesn't add calibration pairs?
+
+  1. One possible reason is that the current pair is too close to previously collected data. In that case, the current data is not accepted.
+  2. The timestamps of the lidar and camera are not synchronized, this can be checked with `ros2 topic echo [topic_name]`. Setting the parameter `use_receive_time` to `True` might help to solve the issue.
+  3. The detections are not stable enough (detections don’t converge).
+
+- Why the UI doesn't launch?
+
+  1. Check with `ros2 node list` if the relevant nodes have been started. It is possible that the provided parameters don’t match any of the valid arguments
+  2. If the UI crashes (see the console), it is probably due to bad PySide installation, invalid intrinsic parameters, or invalid extrinsic parameters.
+  3. The timestamps of the lidar and camera are not synchronized.
+
+- Why the reprojection errors are so high
+
+  1. Check whether the intrinsic parameters are correct.
+
+- Why the reprojection error increases when more data is collected?
+
+  1. When there are few samples, the model will fit the available data the best it can, even in the presence of noise (over-fitting). The more data is collected, the error may increase to a certain extent, but that is the model trying to fit all the data, being unable to fit the noise. However, it should reach a more-or-less table peak with about 10-15 pairs (depending on the data collection pattern/sampling).
+
+- Why reprojection error does not seem low enough?
+
+  1. The intrinsics may not be accurate, thus limiting the performance of the method.
+  2. The boards are not appropriate (are bent).
+  3. The boards moved too much while calibrating.
+  4. The lidar detections were not very good. Try collecting detections in areas where there is more resolution.
```

Comment:
Due to the structure you chose I do not know how to fit this in, but to avoid confusing the filtering algorithm, we require that the users moves decisively from one place to the other, not stopping and starting mid way.

Depending on the mount and the ground surface, the tags can oscillate quite a bit, which can be detrimental to the calibration process. In this case, the user can stop the tag with its hand. This should also be mentioned somewhere in this document

Reply:
Thanks! fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/32dd61431d3beb3d84ecb085cd246042de8efb09